### ISSUE TITLE
Optimize api/v2/events/ N+1

### ISSUE THREAD

**Issue by DeD1rk** [svthalia/concrexit#3586]
### What?
https://thalia.nu/api/v2/events/ is slow, probably at least partially because of https://thalia.sentry.io/issues/4580718277/. We should see if we can get rid of that N+1 issue (and possibly any other occurrences of the same thing.

### Why?
More performance = more happy. It might help with high server load as well.

### How?
Probably some prefetching or annotation.

**Comment by ColonelPhantom** [comment]
I'm having some difficulty reproducing this; I think it only happens in certain edge cases? And I don't remember the exact cases when it happens. Maybe it's related to needing to be in the queue? On Sentry it seems like the queries are related to activemembers.

**Comment by DeD1rk** [comment]
The user may need to be in (organizing) committees?

**Comment by DeD1rk** [comment]
I suspect it's something with gathering permissions that you get for being organizer

**Comment by ColonelPhantom** [comment]
Yeah, probably. It's unfortunate that unlike DJDT, Sentry does not give a backtrace saying where the query originated.

**Comment by ColonelPhantom** [comment]
> I suspect it's something with gathering permissions that you get for being organizer

This is correct. I managed to get a backtrace by creating a non-superuser who is an active member:

```
/home/quinten/code/thalia/concrexit/website/events/api/v2/serializers/event.py in _user_permissions(110)
  return services.event_permissions(member, instance, registration_prefetch=True)

/home/quinten/code/thalia/concrexit/website/events/services.py in event_permissions(184)
  "manage_event": is_organiser(member, event),

/home/quinten/code/thalia/concrexit/website/events/services.py in is_organiser(278)
  .exists()
  ```

The duplicated query is quite ugly as it also involves a big JOIN. I already managed to get rid of most of it with the following code:

```
            organisers = set(event.organisers.all())
            member_groups = set(member.get_member_groups())
            return len(organisers.intersection(member_groups)) > 0
```
but I'm not sure that's optimal either, and as-is it re-retrieves the `get_member_groups` so that would need to be optimized too. For that, I'm not sure about the correct way to do so , since it is too complex to be trivially prefetchable. Maybe the ugly 'named prefetch' thing needs to be done again?
